source/daplink/interface/main.c

```diff
@@ -104,6 +105,43 @@ static main_led_state_t msc_led_state = MAIN_LED_FLASH;
 main_usb_connect_t usb_state;
 static bool usb_test_mode = false;
 
+__WEAK void board_30ms_hook(void)
+{
+
+}
+
+__WEAK void handle_reset_button(void)
+{
+	// button state
+    static uint8_t reset_pressed = 0;
+
+    // handle reset button without eventing
+    if (!reset_pressed && gpio_get_reset_btn_fwrd()) {
+#ifdef DRAG_N_DROP_SUPPORT
+        if (!flash_intf_target->flash_busy()) //added checking if flashing on target is in progress
+#endif
+        {
+            // Reset button pressed
+            target_set_state(RESET_HOLD);
+            reset_pressed = 1;
+        }
+    } else if (reset_pressed && !gpio_get_reset_btn_fwrd()) {
+        // Reset button released
+        target_set_state(RESET_RUN);
+        reset_pressed = 0;
+    }
+}
+
+__WEAK void board_handle_powerdown()
+{
+    // TODO: put the interface chip in sleep mode
+}
```

Comment:
Just noticed that the `while(1);` from `main()` is missing here.

It's probably ok if the main loop keeps running, and might actually be better: RTX will block on the `osThreadFlagsWait()` call and enter the idle thread which will WFI. 

But has this been tested on a build without the micro:bit override of this hook?

Reply:
If I remember correctly I initially had the while(1); in the weak function, but this had strange behavior where the code wasn't executing correctly in the "strong" custom function. I think it had to do with how the compiler optimizations were working. Something about the optimizations being done per file, without knowing that there was a "strong" function definition elsewhere. That and the combination of having an endless loop in the weak function, made the optimizations assume wrong things.

I don't see any build that uses the main_powerdown_event() to trigger this event, so the while(1) was probably a placeholder and there's no impact by removing it, I think.

Reply:
Ok, cool. Nothing to worry about then. Thanks for the quick answer!